src/main/java/com/notably/model/ModelManager.java

```diff
@@ -4,24 +4,31 @@
 import static java.util.Objects.requireNonNull;
 
 import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
 import java.util.function.Predicate;
 import java.util.logging.Logger;
 
 import com.notably.commons.core.GuiSettings;
 import com.notably.commons.core.LogsCenter;
+import com.notably.model.suggestion.SuggestionItem;
+import com.notably.model.suggestion.SuggestionModel;
 
+import javafx.beans.property.Property;
 import javafx.collections.ObservableList;
 import javafx.collections.transformation.FilteredList;
 
+// TODO: to update according to our Model classes.
 /**
  * Represents the in-memory model of the address book data.
  */
 public class ModelManager implements Model {
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
-    private final AddressBook addressBook;
-    private final UserPrefs userPrefs;
-    private final FilteredList<Object> filteredPersons;
+    private AddressBook addressBook;
+    private UserPrefs userPrefs;
+    private FilteredList<Object> filteredPersons;
+    private SuggestionModel suggestionModel;
```

Comment:
Any particular reasons why you removed the `final` keywords?

Reply:
For now I'm following the javafx-todo `Model Manager` code, so I created another constructor which takes in `SuggestionModel` as parameter. This constructor will take in the other model classes as parameters as well in the end. However, if I don't change the addressbook attributes to non-final, it will throw me an error bcs of no initialization. If I need to refactor the whole `ModelManager` code to accommodate the new constructor, I need to change its usage in `MainApp.java` too.

So for the time being, in order to allow me not to change ModelManager's usage in `MainApp.java`, I just removed the `final` keywords.

Do you have an advice on how I should do about this?

Reply:
you cant use final because apparently all the constructors (including overloaded ones such as the new one you defined) need to initialise the value of final variables. In your case the new constructor does not do this for the other variables, hence you get the warning "may not have been initialised". 

For now, I think can remove.

Reply:
I see, thanks @HemanshuGandhi !